### ISSUE TITLE
Add pages that list all known django-component plugins and components

### ISSUE THREAD

**Issue by JuroOravec** [django-components/django-components#805]
Offshoot from https://github.com/EmilStenstrom/django-components/discussions/790, where the suggestion was to:
1. Assuming that pip / PyPI is currently the best way to distribute components built on top of django-components.

2. We'd agree on a prefix that PyPI packages may use to identify them as a plugin or component(s) for django-components
    - E.g. `djc-component-` for components and `djc-plugin-` for plugins.
      - So my icon component package that wraps icons from HeroIcons.com  could be named `djc-component-heroicons`
      
3. We'd add two pages, one to list all found components, and one to list all found plugins.

4. The pages would be constructed similarly to how we construct the API reference:
    1. We'd hook into the mkdocs build flow using `gen-files` plugin, same as we do with `reference.py`
    2. Inside the hook, we'd query PyPI to find all packages that start with `djc-component-` and `djc-plugin-`
    3. We'd render these packages into markdown / HTML, showing: package name, description, author name, maybe downloads / stars (depending on what data is available).
       - See [MeteorJS](https://atmospherejs.com/), [NPM](https://www.npmjs.com/search?q=hello), or [PyPI](https://pypi.org/search/?q=django) for inspiration

**Comment by EmilStenstrom** [comment]
I don't think this is a good idea. Forcing people to rename their library to fit our format feels convoluted. Maybe we could instead require a custom tag be added?

**Comment by JuroOravec** [comment]
I had a look, and using names feels like the safest approach here. The problem is that PyPI deprecated the API for searching, because it was being used more than they could handle.

See
- https://pip.pypa.io/en/stable/cli/pip_search/#description
- https://stackoverflow.com/a/65584476/9788634
- https://stackoverflow.com/a/61787849/9788634

Which is a shame, because tags ARE part of the full-text search - e.g. on the website if I searched for "djc-plugin-", it would return packages that have that that text either in name, description, or tags. See https://pypi.org/search/?q=%22djc-plugin%22&o=.

But so, the way to query by tags would involve either using a deprecated API (`pip search`), or scraping data from PyPI website. Neither sound great.
- Note: In either case, for each potential match we'd need to get the package's full metadata to confirm the tags, but that's that's not an issue, there's a working JSON API for that https://docs.pypi.org/api/json/#get-a-project.

On the other hand, searching by name is much simpler. Again, not ideal, because the search API is deprecated, but at least supported by PyPI:
1. Download the index of all package names known to PyPI from https://pypi.org/simple/ (also available as JSON)
2. Take only those projects whose name starts with `djc-plugin-`.

---

Nevertheless, the name prefix would be optional. People could still define packages with different names, but then it would be on them to tell us about it.

IMO whichever method we'd use to get the list of packages, we'd still want to be able to:
- Add extra packages manually in case they don't fit the format
- Remove packages manually, in case there are false positives, or someone exploits it (e.g. makes a fake package for advertising space)

This approach is nothing new, ESLint does this, they use `eslint-config-` for configs and `eslint-plugin-` for plugins (see [docs](https://eslint.org/docs/latest/extend/plugins), [list of plugins](https://github.com/dustinspecker/awesome-eslint)). I remember also Gridsome doing it https://gridsome.org/plugins/, and few others.

**Comment by JuroOravec** [comment]
Implemented in Citry - [Extensions](https://citry.dev/community/extensions/) and [Libraries](https://citry.dev/community/ui-libraries/)

Entries are managed manually. no automatic scanning of PyPI, and no requirements on naming of labels.
